src/lib/CRSF/CRSF.cpp

```diff
@@ -383,6 +383,37 @@ void ICACHE_RAM_ATTR CRSF::sendSyncPacketToTX(void *pvParameters) // in values i
 
             this->_dev->write(outBuffer, RCframeLength + 4);
         }
+
+        void ICACHE_RAM_ATTR CRSF::sendMSPFrameToFC(mspPacket_t* packet)
+        {
+            // TODO: This currently only supports single MSP packets per cmd
+            // To support longer packets we need to re-write this to allow packet splitting
+
+            uint8_t outBuffer[ENCAPSULATED_MSP_FRAME_LEN + CRSF_FRAME_LENGTH_EXT_TYPE_CRC + CRSF_FRAME_NOT_COUNTED_BYTES] = {0};
+
+            // CRSF extended frame header
+            outBuffer[0] = CRSF_ADDRESS_BROADCAST; // address
+            outBuffer[1] = ENCAPSULATED_MSP_FRAME_LEN + CRSF_FRAME_LENGTH_EXT_TYPE_CRC; // length
+            outBuffer[2] = CRSF_FRAMETYPE_MSP_WRITE; // packet type
+            outBuffer[3] = CRSF_ADDRESS_FLIGHT_CONTROLLER; // destination
+            outBuffer[4] = CRSF_ADDRESS_RADIO_TRANSMITTER; // origin
+            
+            // Encapsulated MSP payload
+            outBuffer[5] = 0x30; // header
+            outBuffer[6] = packet->payloadSize; // mspPayloadSize
+            outBuffer[7] = packet->function; // packet->cmd
+            for (uint8_t i = 0; i < ENCAPSULATED_MSP_PAYLOAD_SIZE; ++i) {
+                // copy packet payload into outBuffer and pad with zeros where required
+                outBuffer[8 + i] = i < packet->payloadSize ? packet->payload[i] : 0;
+            }
+            // Encapsulated MSP crc
+            outBuffer[12] =  CalcCRCMsp(&outBuffer[6], ENCAPSULATED_MSP_FRAME_LEN - 2);
```

Comment:
The 12 and 13 constants are a bit of a hostage to fortune. Mighe be worth calculating a constant for the total length around line 391 before allocating the array, and then use total length -2 and total length -1 as the indices here?

Reply:
Changed based on suggestions.